Basics/Git.md

```diff
@@ -4,50 +4,130 @@
+3. Capitalize the subject line
+4. Do not end the subject line with a period
+5. Use the imperative mood in the subject line
+6. Wrap the body at 72 characters
+7. Use the body to explain what and why vs. how
+8. You can find more details about those in [this blog post][3].
 
-## 3. GitFlow for small projects
+There is also a [git hook][4] available to validate your commit message.
 
-In order to keep things rolling faster, small projects also use GitFlow, but to a very limited extent.
+## 6. Pull requests
 
-For one-man projects, you will have only the **master** and **develop** branches, and will commit your code directly to develop, while the master stays the same.
+The preferred way of merging a branch back into main is by creating a pull request and assigning it to a colleague for review. At least one person should review any code that is going to be merged (reviewer on PR).
 
-If there is more than one developer, you'll need a standard GitFlow model.
+Each PR should have a description. Link to the Productive/Jira task is minimum for description. It would be great to include a link to design (if applicable) and describe the changes that PR introduces. You can find a good read on what the description should contain [here][5].
 
-## 4. Tools (optional)
+PR should have an assignee, which is usually a person who created a PR or the person who will be responsible for merging it. 
 
-Although you don’t need to use any tools with GitFlow workflow, and may prefer sticking to the workflow manually, you still have an option to add commands to your console/terminal for commonly used workflows.
+Master and/or develop branches should always be marked as protected on Github in order to prevent direct push to those branches. Next options should be checked on branch protection rules page on Github:
+* ***Require pull request reviews before merging***
+![Pull request][image-2]
+* ***Require status check to pass before merging (this includes any CI and Sonarcloud if used)***
+![Status chec][image-3]
+* ***Require branches to be up to date before merging***
+![Branches][image-4]
 
-If you want to use Git with Gitflow commands and terms, you might find the [git-flow][2] tool useful.
+## 7. Mobile specific
 
-[1]:	https://www.atlassian.com/git/tutorials/comparing-workflows#gitflow-workflow
-[2]:	https://github.com/nvie/gitflow
+Above mentioned rules should be the minimum for all platforms. Few more are specific and applicable only for mobile platforms. If you cannot apply some of the mentioned rules to your project, please talk with another platform and try to find a viable solution for both of you.
+
+### Git Flow/Infinum Flow
+
+We’ll be using the Git Flow standard as a base for git branching with some modifications. More about Git Flow [here][6].
+
+#### Branches
+We’ll be using **master** as the main branch. Since we’ll be using tags and optionally **release** branches for deployment, **develop** branch becomes redundant.
```

Comment:
```suggestion
We’ll be using **master** as the main branch. Since we’ll be using tags and optionally **release** branches for deployment, **develop** branch becomes redundant and therefore should not be used in the project.
```